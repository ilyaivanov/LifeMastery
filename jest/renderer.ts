import TestRenderer from "react-test-renderer";

const renderer = (element: any) => {
  const rendered = TestRenderer.create(element);
  return {
    text: (id: string) => findTextById(rendered.root, verifyNotNull(id)),
    touch: (id: string) => touchById(rendered.root, verifyNotNull(id)),

    setState: (state: any) => rendered.root.instance.setState(state),

    toJSON: () => rendered.toJSON()
  };
};

const verifyNotNull = (id: string) => {
  if (!id) throw new Error("You need to provide a valid id, but id was: " + id);
  return id;
};

const findTextById = (instance: any, id: string) =>
  instance.findByProps({ testID: id }).props.children;

const touchById = (instance: any, id: string) =>
  instance.findByProps({ testID: id }).props.onPress();

export default renderer;
export type Rendered = ReturnType<typeof renderer>;
