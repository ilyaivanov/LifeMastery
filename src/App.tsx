import React, { Component } from "react";
import { Button, StyleSheet, Text, View } from "react-native";
import { showSecondScreen, goBack } from "./navigation";

interface Props {
  componentId: string
}

export default class App extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <Text testID="PageTitle" style={styles.welcome}>First Screen</Text>
        <Button testID="GoNext" onPress={() => showSecondScreen(this.props.componentId)} title="Go next!" />
      </View>
    );
  }
}

export class SecondScreen extends Component<Props> {
  render() {
    return (
      <View style={[styles.container, { backgroundColor: 'grey' }]}>
        <Text testID="PageTitle" style={styles.welcome}>Second Screen</Text>
        <Button testID="GoBack" onPress={() => goBack(this.props.componentId)} title="Go back!" />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10
  },
});
