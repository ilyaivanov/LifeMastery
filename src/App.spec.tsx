import "react-native";
import React from "react";
import App from "./App";
import renderer, {Rendered} from "../jest/renderer";

describe('When opening App', () => {
    let app: Rendered;
    beforeEach(() => {
        app = renderer(<App/>);

    });

    it("should renders correctly", () => {
        expect(app.toJSON()).toMatchSnapshot();
    });

    it("by default increment should be 0", () => {
        expect(app.text("countLabel")).toEqual(0);
    });

    describe("when touching a button", () => {
        it("increment should be 1", () => {
            const tree = renderer(<App/>);
            tree.touch("incrementButton");
            expect(tree.text("countLabel")).toEqual(1);
        });
    });

});



