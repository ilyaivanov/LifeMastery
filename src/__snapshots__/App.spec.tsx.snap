// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When opening App should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
    testID="countLabel"
  >
    0
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="incrementButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment
      </Text>
    </View>
  </View>
</View>
`;
